cmd/upgrade.go

```diff
@@ -110,19 +109,22 @@ func (uc *upgradeCmd) validate(cmd *cobra.Command, args []string) {
 		log.Fatalf("error parsing the api model: %s", err.Error())
 	}
 
-	if _, err := os.Stat(uc.upgradeModelFile); os.IsNotExist(err) {
+	if _, err = os.Stat(uc.upgradeModelFile); os.IsNotExist(err) {
 		log.Fatalf("specified upgrade model file does not exist (%s)", uc.upgradeModelFile)
 	}
 
-	upgradeapiloader := &api.UpgradeApiloader{
-		Translator: &i18n.Translator{
-			Locale: uc.locale,
-		},
+	// acs-engine command line uses vlabs API
+	contents, err := ioutil.ReadFile(uc.upgradeModelFile)
+	if err != nil {
+		log.Fatalf("error reading file %s: %s", uc.upgradeModelFile, err.Error())
 	}
-	uc.upgradeContainerService, uc.upgradeAPIVersion, err = upgradeapiloader.LoadUpgradeContainerServiceFromFile(uc.upgradeModelFile)
+	upgradeModel, err := apiloader.LoadUpgradeContainerService(contents, vlabs.APIVersion, true)
 	if err != nil {
-		log.Fatalf("error parsing the upgrade api model: %s", err.Error())
+		log.Fatalf("error loading UpgradeContainerService: %s", err.Error())
 	}
+	// set GoalState
+	uc.containerService.Properties.OrchestratorProfile.OrchestratorRelease = upgradeModel.OrchestratorRelease
+	uc.containerService.Properties.OrchestratorProfile.OrchestratorVersion = upgradeModel.OrchestratorVersion
```

Comment:
IIRC the decision was that customer was only going to provide in the input OrchestratorRelease and OrchestratorVersion  would be determined by the upgrade operation.